cmd/launchBackup.go

```diff
@@ -0,0 +1,234 @@
+		err = executeArgs(bashArgs, path[0], "containers")
+		if err != nil {
+			log.Warn(err)
+		}
+	*/
+	// cluster backup
+	err = executeArgs(BackupScript, path[1], "etcd-cluster")
+	if err != nil {
+		return err
+	}
+
+	// etc back up
+	etcExcludeArgs := fmt.Sprintf("cat /etc/tmpfiles.d/* | sed 's/#.*//' | awk '{print $2}' | grep '^/etc/' | sed 's#^/etc/##' > %s/etc.exclude.list; echo '.updated' >> %s/etc.exclude.list; echo 'kubernetes/manifests' >> %s/etc.exclude.list", BackupPath, BackupPath, BackupPath)
+	err = executeArgs(etcExcludeArgs, BackupPath, "etc-exclude-list")
+	if err != nil {
+		log.Error(err)
+	}
+
+	// usrlocal backup
+	etcArgs := fmt.Sprintf("rsync -a %s %s", etc, path[2])
+	err = executeArgs(etcArgs, path[2], etc)
+	if err != nil {
+		log.Error(err)
+	}
+
+	usrlocalArgs := fmt.Sprintf("rsync -a %s %s", usrlocal, path[3])
+	err = executeArgs(usrlocalArgs, path[3], usrlocal)
+	if err != nil {
+		log.Error(err)
+	}
+
+	kubeletArgs := fmt.Sprintf("rsync -a %s %s", kubelet, path[4])
+	err = executeArgs(kubeletArgs, path[4], kubelet)
+	if err != nil {
+		log.Error(err)
+	}
+
+	log.Info(strings.Repeat("-", 60))
+	log.Info("backup has succesfully finished ...")
+
+	return nil
+
+}
+
+// cleanup deletes all old sub diectories and files in the recovery partition
+// returns: 			error
+func cleanup(path string) error {
+	//change root directory to /host
+	if err := syscall.Chroot(host); err != nil {
+		log.Println("Couldn't do chroot to %s", host)
+		return nil
+	}
+
+	log.Info(strings.Repeat("-", 60))
+	log.Info("Cleaning up old contents and ostree deployment started ...")
+	log.Info(strings.Repeat("-", 60))
+	// Cleanup previous backups
+	dir, _ := os.Open(path)
+	subDir, _ := dir.Readdir(0)
+
+	// Loop over the directory's files.
+	for index := range subDir {
+		fileNames := subDir[index]
+
+		// Get name of file and its full path.
+		name := fileNames.Name()
+		fullPath := path + "/" + name
+		log.Info("\nfullpath: ", fullPath)
+
+		if subDir[index].Name() != "extras.tgz" {
+			// Remove the file.
+			err := os.RemoveAll(fullPath)
+			if err != nil {
+				log.Error(err)
+				return err
+			}
+		}
+	}
+	log.Info("Old directories deleted with contents")
+
+	// ostree undeploy of previous deployments
+	ostreeClean := "while :; do ostree admin undeploy 1 || break; done"
+	err := executeArgs(ostreeClean, "backup-time", "ostree-cleaning")
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+	return nil
+}
+
+//createDir creates new sub-directories where backup will be stored
+// returns:  slice of filepath ([]string) error
+func createDir(path string) ([]string, error) {
+	//create backup folders
+	newPath := make([]string, len(folders))
+	os.Chdir(path)
+
+	for i := 0; i < len(folders); i++ {
+		err := os.Mkdir(folders[i], os.ModePerm)
```

Comment:
os.ModePerm is 0x777, I believe. The perms here should probably be 0x700, or at least 0x755, I think.

Reply:
sure, 0755 is better as this modeperm is widely used in webserver

Reply:
@donpenney Removed as suggested in commit 7b894ea